source/Fleet.h

```diff
@@ -82,13 +89,14 @@ class Fleet {
 
 
 private:
+	bool hasConditions = true;
 	std::string fleetName;
 	const Government *government = nullptr;
 	const Phrase *names = nullptr;
 	const Phrase *fighterNames = nullptr;
-	WeightedList<Variant> variants;
+	WeightedList<Variant,WeightType> variants;
 	// The number of different items the ships in this fleet will carry in cargo.
-	int cargo = 3;
+	RValue<int> cargo;
```

Comment:
What advantages do you see wrapping the cargo int in this RValue class give you here?

Reply:
For anything to be configurable via a condition, it needs an RValue so you can map it back to the condition variable name it came from. Luckily, cargo is an int, not a double, so it can be configured via condition.